engine/tests/ElggCoreFilestoreTest.php

```diff
@@ -97,6 +97,34 @@ function testElggGetFileSimpletype() {
 		}
 	}
 
+	function testDetectMimeType() {
+
+		$user = $this->createTestUser();
+
+		$file = new \ElggFile();
+		$file->owner_guid = $user->guid;
+		$file->setFilename('testing/filestore.txt');
+		$file->open('write');
+		$file->write('Testing!');
+		$file->close();
+
+		$mime = $file->detectMimeType(null, 'text/plain');
+
+		// mime should not be null if default is set
+		$this->assertTrue(isset($mime));
+
+		// mime of a file object should match mime of a file path that represents this file on filestore
+		$resource_mime = $file->detectMimeType($file->getFilenameOnFilestore(), 'text/plain');
+		$this->assertIdentical($mime, $resource_mime);
+
+		// method output should not differ between a static and a concrete call if the file path is set
+		$resource_mime_static = ElggFile::detectMimeType($file->getFilenameOnFilestore(), 'text/plain');
```

Comment:
PHPUnit raises warnings to exceptions, so I think you'll need something like:

before: `error_reporting(E_ALL & ~E_STRICT);`
after: `error_reporting(E_ALL);`